crates/driver/src/print_build.rs

```diff
@@ -248,127 +248,271 @@ where
     };
 
     let target_folder = opt.working_directory.join(&element);
-    let target_name = target_folder
-        .file_name()
-        .unwrap()
-        .to_str()
-        .unwrap()
-        .to_string();
+    let base_name = to_file_name(&target_folder);
+    let mut t = TargetEntries {
+        entries: Vec::default(),
+    };
+    for graph_node in graph_nodes {
+        let node_file = opt.working_directory.join(&graph_node.node_label);
+        let node_file_name = to_file_name(&node_file);
+        let target_name = if !opt.no_aggregate_source {
+            base_name.clone()
+        } else {
+            node_file_name.replace(".", "_")
+        };
+        let mut extra_kv_pairs: HashMap<String, Vec<String>> = HashMap::default();
+
+        if !graph_node.dependencies.is_empty() {
```

Comment:
nit: can we reduce repetition:
```rust
fn add_non_empty(key: String, labels: Vector<String>, mut& extra_kv_pairs: Hash<String, Vector<String>>) {
  if !labels.is_empty() {
    let vals = labels.iter()
                .map(|e| to_label(&opt, e))
                .collect();
    extra_kv_pairs.insert(key, vals)
  }
}
```
then call it twice?

Reply:
Addressed in https://github.com/bazeltools/bzl-gen-build/pull/117/commits/b81b4341e7af6afbe90894aa28e682cdcc3022d4